icu4c/source/i18n/unicode/numberformatter.h

```diff
@@ -1512,6 +1512,9 @@ class U_I18N_API NumberFormatterSettings {
      *
      * Pass this method any instance of {@link MeasureUnit}. For units of measure:
      *
+     * NOTE:    If the `usage` is set, the output unit **will be changed**
+     *          according to the `usage`, `locale` and `unit` values.
```

Comment:
We also need to decide what will happen with the ".perUnit()" functionality below. Do we want to support .unit("foo").perUnit("bar") to mean the same thing as .unit("foo-per-bar")?

Reply:
> Do we want to support .unit("foo").perUnit("bar") to mean the same thing as .unit("foo-per-bar")?

Yes; that's already what happens.  See the unit tests.

Reply:
It would be good to add this behavior to the API docs.

Reply:
I've not touched this suggestion yet. I would want to add unit tests to also prove what happens when one has .unit(MIXED).perUnit(MIXED) (an error, surely), or COMPOUND, or a permutation of MIXED and COMPOUND.

Reply:
If you'd like, I could add this as a TODO somewhere? (Once glue code is implemented, I want to do end-to-end verification of the public API as documented in the header file.)

Reply:
The semantics of unit/perUnit are orthogonal to usage().  You can resolve this thread; I didn't want to resolve it since I didn't open it originally.